modules/lvm2/storagedlinuxlogicalvolume.c

```diff
@@ -1030,6 +1018,208 @@ handle_create_snapshot (StoragedLogicalVolume *_volume,
   return TRUE;
 }
 
+static gboolean
+handle_cache_attach (StoragedLogicalVolume  *volume_,
+                     GDBusMethodInvocation  *invocation,
+                     const gchar            *cache_name,
+                     GVariant               *options)
+{
+#ifndef HAVE_LVMCACHE
+
+  g_dbus_method_invocation_return_error (invocation,
+                                         STORAGED_ERROR,
+                                         STORAGED_ERROR_FAILED,
+                                         N_("LVMCache not enabled at compile time."));
+  return TRUE;
+
+#else
+
+  GError *error = NULL;
+  StoragedLinuxLogicalVolume *volume = STORAGED_LINUX_LOGICAL_VOLUME (volume_);
+  StoragedLinuxLogicalVolumeObject *object = NULL;
+  StoragedDaemon *daemon;
+  uid_t caller_uid;
+  StoragedLinuxVolumeGroupObject *group_object;
+  GString *cmd;
+  gchar *escaped_group_name = NULL;
+  gchar *escaped_origin_name = NULL;
+  gchar *error_message;
+
+  object = storaged_daemon_util_dup_object (volume, &error);
+  if (object == NULL)
+    {
+      g_dbus_method_invocation_take_error (invocation, error);
+      goto out;
+    }
+
+
+  daemon = storaged_linux_logical_volume_object_get_daemon (object);
+
+  if (!storaged_daemon_util_get_caller_uid_sync (daemon,
+                                                 invocation,
+                                                 NULL /* GCancellable */,
+                                                 &caller_uid,
+                                                 NULL,
+                                                 NULL,
+                                                 &error))
+    {
+      g_dbus_method_invocation_take_error (invocation, error);
+      goto out;
+    }
+
+  STORAGED_DAEMON_CHECK_AUTHORIZATION (daemon,
+                                       STORAGED_OBJECT (object),
+                                       lvm2_policy_action_id,
+                                       options,
+                                       N_("Authentication is required to convert logica volume to cache"),
+                                       invocation);
+
+  group_object = storaged_linux_logical_volume_object_get_volume_group (object);
+
+  escaped_group_name = storaged_daemon_util_escape_and_quote (storaged_linux_volume_group_object_get_name (group_object));
+  escaped_origin_name = storaged_daemon_util_escape_and_quote (storaged_linux_logical_volume_object_get_name (object));
+  cache_name = storaged_daemon_util_escape_and_quote (cache_name);
```

Comment:
Memory leak is present here.

Suggestion: Use a different variable for this, `cache_name` and `g_free()` it afterwards.

Because:
```
gchar *
storaged_daemon_util_escape_and_quote (const gchar *str)
{
  GString *s;

  g_return_val_if_fail (str != NULL, NULL);

  s = g_string_new ("\"");
  escaper (s, str);
  g_string_append_c (s, '"');

  return g_string_free (s, FALSE);
}
```

Sorry for not catching this earlier.